tv_apps/smart-system/js/splash_screen.js

```diff
@@ -0,0 +1,62 @@
+/* global PreviewWindow */
+
+(function(exports) {
+  'use strict';
+
+  var _id = 0;
+
+  var SplashScreen = function SplashScreen(app) {
+    this.app = app;
+    this.containerElement = app.element;
+
+    if (this.app instanceof PreviewWindow) {
+      this.instanceID = _id++;
+      this.render();
+    }
+  };
+
+  SplashScreen.prototype = Object.create(window.BaseUI.prototype);
+
+  SplashScreen.prototype.CLASS_NAME = 'SplashScreen';
+  SplashScreen.prototype.EVENT_PREFIX = 'splash-screen-';
+
+  SplashScreen.prototype.view = function ss_view() {
+    return `<div id="${this.CLASS_NAME + this.instanceID}"
+                 class="splash-screen hidden"></div>`;
+  };
+
+  SplashScreen.prototype._fetchElements = function ss__fetchElements() {
+    this.element = document.getElementById(this.CLASS_NAME + this.instanceID);
+  };
+
+  SplashScreen.prototype._registerEvents = function ss__registerEvents() {
+    this.app.element.addEventListener('_loading', this);
+    this.app.element.addEventListener('_loaded', this);
+  };
+
+  SplashScreen.prototype.handleEvent = function ss_handleEvent(evt) {
+    switch (evt.type) {
+      case '_loading':
+        if (this.app instanceof PreviewWindow) {
+          this.show();
+        }
+        break;
+
+      case '_loaded':
+        if (this.app instanceof PreviewWindow) {
+          this.hide();
```

Comment:
The event `_loading` and `_loaded` will be triggered every time when the app loads a new url. You needs to remove the event handlers (and the DOM elements) if you don't want to see this splash screen twice.